### PR TITLE
Backend dep bump for 50

### LINKED ISSUE #40354: Start of 50 Cycle Routine Clojure Dependency Bump
[Notion doc about dependency bumping](https://www.notion.so/metabase/Dependency-Bumping-8ab452bbfc1f4131addd1653235c1b0f?pvs=4)

Subtasks:

- [x] Update all minor and patch versions and test. Revert breakage.
- [x] Update minor versions in modules
- [x] Determine whether we want to upgrade the Mongo driver to version 5
- [x] Anything else, TBD

### PR DESCRIPTION
**Pull request by tsmacdonald**
Fixes #40354 .

Passes CI and stood up to some light smoke-testing, so I think it's time to get it in master / on stats and let it get properly flexed.

### CODE REVIEW

File: deps.edn [hunk 1]
```diff
@@ -73,13 +73,13 @@
...
-                                             :git/sha "d3210d3"
+  io.github.camsaul/toucan2                 {:mvn/version "1.0.538"}
+  io.github.eerohele/pp                     {:git/tag "2024-01-04.60"           ; super fast pretty-printing library
+                                             :git/sha "a428751"
                                              :git/url "https://github.com/eerohele/pp"}
-  io.github.metabase/macaw                  {:mvn/version "0.1.21"}             ; Parse native SQL queries
+  io.github.metabase/macaw                  {:mvn/version "0.1.23"}             ; Parse native SQL queries
   ;; The 2.X line of Resilience4j requires Java 17, so we cannot upgrade this dependency until that is our minimum JVM version
-  io.github.resilience4j/resilience4j-retry {:mvn/version "1.7.1"}              ; Support for retrying operations
+  io.github.resilience4j/resilience4j-retry {:mvn/version "1.7.1" #_ "must be 1.7.1"} ; Support for retrying operations
```
Comment (tsmacdonald) [inline deps.edn:82]: if you're using one of the auto-upgrade tools it can be easy to glaze over the comment on L81

File: deps.edn [hunk 2]
```diff
@@ -328,42 +329,24 @@
...
-               "dev" "src" "enterprise/backend/src"]}
-
-  ;; Generate a single file's docs with
-  ;;   clojure -M:marginalia/one src/metabase/some/namespace.clj
-  ;; which creates `backend-docs/metabase.some.namespace.html`
-  :marginalia/one
-  {:extra-deps
-   {com.github.tsmacdonald/marginalia {:mvn/version "0.9.2"}}
-   :main-opts ["-m" "marginalia.main" "-m" "-n" "Metabase" "-d" "backend-docs" "-D"
-               "The simplest, fastest way to get business intelligence and analytics to everyone in your company 😋"]}
```
Comment (tsmacdonald) [inline deps.edn:347]: We do it [a different way](https://github.com/metabase-dev-docs/metabase-dev-docs.github.io/) now

File: deps.edn [hunk 3]
```diff
@@ -108,72 +108,75 @@
...
+  org.clojure/math.numeric-tower            {:mvn/version "0.1.0"}              ; math functions like `ceil`
+  org.clojure/tools.cli                     {:mvn/version "1.1.230"}            ; command-line argument parsing
+  org.clojure/tools.logging                 {:mvn/version "1.3.0"}              ; logging framework
+  org.clojure/tools.macro                   {:mvn/version "0.2.0"}              ; local macros
+  org.clojure/tools.namespace               {:mvn/version "1.5.0"}
+  org.clojure/tools.reader                  {:mvn/version "1.4.1"}
+  org.clojure/tools.trace                   {:mvn/version "0.8.0"}              ; function tracing
+  ;; v. 12 of jetty-server triggers "a org/eclipse/jetty/io/EofException has been compiled by a more recent version of
+  ;; the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions
+  ;; up to 55.0"
```
Comment (tsmacdonald) [inline deps.edn:157]: image

https://github.com/metabase/metabase/actions/runs/8360468213/job/22886205792?pr=40372

File: deps.edn [hunk 4]
```diff
@@ -36,15 +36,15 @@
                                              :exclusions  [it.unimi.dsi/fastutil
                                                            org.slf4j/slf4j-api]}
   com.draines/postal                        {:mvn/version "2.0.5"}              ; SMTP library
-  com.github.seancorfield/honeysql          {:mvn/version "2.5.1103"}           ; Honey SQL 2. SQL generation from Clojure data maps
+  com.github.seancorfield/honeysql          {:mvn/version "2.6.1126"}           ; Honey SQL 2. SQL generation from Clojure data maps
   com.github.seancorfield/next.jdbc         {:git/url "https://github.com/seancorfield/next-jdbc.git"
                                              :sha     "abd926cde3730087d3729dcea0ce0ab7ef1194f2"} ; for https://github.com/seancorfield/next-jdbc/issues/245; when this makes it to a release we can switch to that.
```
Comment (piranha) [inline deps.edn:41]: Don't want to update this one? :) The fix is merged already, but last time I've tried that (updating next.jdbc and honeysql) I broke master - in 10cdb49db4d5105962e418c5f23f9bdae3309450

**Review by sansolovyov — APPROVED** [review]
Maybe next.jdbc as a separate PR? :)

File: deps.edn (hunk 4 above)
Comment (tsmacdonald) [reply deps.edn]: ah, good call. Confirmed that it's in the newer version, so we should be good tog o

Comment (piranha) [reply deps.edn]: Beware: `be-linter-cloverage` step runs only on master, and that was the one broken when I tried to update this back in November. So maybe it still makes sense to have `next.jdbc` as a separate PR?

Comment (tsmacdonald) [reply deps.edn:41]: fair enough, will tag you in the separate PR in a sec

Comment (tsmacdonald) [reply deps.edn:41]: https://github.com/metabase/metabase/pull/40452

**Pull request merged**